gremlin-go/driver/traversal.go

```diff
@@ -450,3 +450,29 @@ func (tp *textP) And(args ...interface{}) TextPredicate {
 func (tp *textP) Or(args ...interface{}) TextPredicate {
 	return newTextPWithP("or", *tp, args...)
 }
+
+type withOptions struct {
+	Tokens  string
+	None    int32
+	Ids     int32
+	Labels  int32
+	Keys    int32
+	Values  int32
+	All     int32
+	Indexer string
+	List    int32
+	Map     int32
+}
+
+var WithOptions withOptions = withOptions{
```

Comment:
What does this look like from a consuming project?

I worry about the combination of exporting an unexported type with exported internal fields.

Reply:
gremlingo.WithOptions.<Option>, same as other drivers.